server/public/model/channel.go

```diff
@@ -356,6 +356,28 @@ func (o *Channel) GetOtherUserIdForDM(userId string) string {
 	return otherUserId
 }
 
+func (o *Channel) Sanitize() {
+	// the only properties that are not cleared.
+	// Id, TeamId, Type, DisplayName
+	o.CreateAt = 0
+	o.UpdateAt = 0
+	o.DeleteAt = 0
+	o.Name = ""
+	o.Header = ""
+	o.Purpose = ""
+	o.LastPostAt = 0
+	o.TotalMsgCount = 0
+	o.ExtraUpdateAt = 0
+	o.CreatorId = ""
+	o.SchemeId = nil
+	o.Props = nil
+	o.GroupConstrained = nil
+	o.Shared = nil
+	o.TotalMsgCountRoot = 0
+	o.PolicyID = nil
+	o.LastRootPostAt = 0
+}
```

Comment:
This may be overkill, but starting extremely restrictive and only returning what is necessary

Reply:
Alternatively, you could consider having this method create a new channel object, copy the needed fields, and then return it. That way, we wouldn't have to worry about any new fields that are added in the future not being sanitized out

Reply:
That's a great point from @hmhealey